arangod/Aql/AqlValue.cpp

```diff
@@ -1473,11 +1794,29 @@ bool std::equal_to<arangodb::aql::AqlValue>::operator()(arangodb::aql::AqlValue
   if (type != b.type()) {
     return false;
   }
-  if (type == arangodb::aql::AqlValue::VPACK_INLINE) {
-    return arangodb::velocypack::Slice(&a._data.internal[0])
-           .binaryEquals(arangodb::velocypack::Slice(&b._data.internal[0]));
+  switch (type) {
+    case arangodb::aql::AqlValue::VPACK_INLINE:
+      return arangodb::velocypack::Slice(a._data.inlineSliceMeta.slice)
+           .binaryEquals(arangodb::velocypack::Slice(b._data.inlineSliceMeta.slice));
+    case arangodb::aql::AqlValue::VPACK_INLINE_INT48:
+    case arangodb::aql::AqlValue::VPACK_INLINE_UINT48:
+      // equal is equal. sign does not matter. So compare unsigned
+      return a._data.shortNumberMeta.data.uint48.val == b._data.shortNumberMeta.data.uint48.val;
+    case arangodb::aql::AqlValue::VPACK_INLINE_INT64:
+    case arangodb::aql::AqlValue::VPACK_INLINE_UINT64:
+    case arangodb::aql::AqlValue::VPACK_INLINE_DOUBLE:
+      // equal is equal. sign/endianess does not matter
+      return a._data.longNumberMeta.data.intLittleEndian.val == b._data.longNumberMeta.data.intLittleEndian.val;
+      return arangodb::velocypack::Slice(a._data.longNumberMeta.data.slice.slice)
+           .binaryEquals(arangodb::velocypack::Slice(b._data.longNumberMeta.data.slice.slice));
```

Comment:
Something's not right here - why are there two return statements? And the double case probably has to be treated differently.

Reply:
Good catch 😄  Copy&paste is evil